lib/internal/crypto/ec.js

```diff
@@ -166,7 +166,25 @@ async function ecGenerateKey(algorithm, extractable, keyUsages) {
       // Fall through
   }
   return new Promise((resolve, reject) => {
-    generateKeyPair('ec', { namedCurve }, (err, pubKey, privKey) => {
+    let generate;
+    switch (namedCurve) {
+      case 'NODE-ED25519':
+        generate = generateKeyPair.bind(undefined, 'ed25519');
+        break;
+      case 'NODE-ED448':
+        generate = generateKeyPair.bind(undefined, 'ed448');
+        break;
+      case 'NODE-X25519':
+        generate = generateKeyPair.bind(undefined, 'x25519');
+        break;
+      case 'NODE-X448':
+        generate = generateKeyPair.bind(undefined, 'x448');
+        break;
+      default:
+        generate = generateKeyPair.bind(undefined, 'ec', { namedCurve });
+        break;
+    }
+    generate((err, pubKey, privKey) => {
```

Comment:
I'd strongly advise to not use `bind` here due to the complexity overhead and my feeling that some members would want to replace all of the calls to `bind` with the relevant primordials, adding even more complexity. You could either replace these with arrow functions of the form `(cb) => generateKeyPair('x448', cb)`, or break it down to the parameters like this:

```suggestion
    let genKeyType, genOpts = {};
    switch (namedCurve) {
      case 'NODE-ED25519':
        genKeyType = 'ed25519';
        break;
      case 'NODE-ED448':
        genKeyType = 'ed448';
        break;
      case 'NODE-X25519':
        genKeyType = 'x25519';
        break;
      case 'NODE-X448':
        genKeyType = 'x448';
        break;
      default:
        genKeyType = 'ec';
        genOpts = { namedCurve };
        break;
    }
    generateKeyPair(genKeyType, genOpts, (err, pubKey, privKey) => {
```